frameworks/FEDOT/exec_ts.py

```diff
@@ -0,0 +1,151 @@
+
+    scoring_metric = get_fedot_metrics(config)
+
+    training_params = {"preset": "best_quality", "n_jobs": config.cores}
+    training_params.update({k: v for k, v in config.framework_params.items() if not k.startswith('_')})
+    n_jobs = training_params["n_jobs"]
+
+    log.info('Running FEDOT with a maximum time of %ss on %s cores, optimizing %s.',
+             config.max_runtime_seconds, n_jobs, scoring_metric)
+    runtime_min = config.max_runtime_seconds / 60
+
+    task = Task(
+        TaskTypesEnum.ts_forecasting,
+        TsForecastingParams(forecast_length=dataset.forecast_horizon_in_steps)
+    )
+
+    train_df, test_df = load_timeseries_dataset(dataset)
+    id_column = dataset.id_column
+
+    log.info('Predicting on the test set.')
+    training_duration, predict_duration = 0, 0
+    models_count = 0
+    truth_only = test_df[dataset.target].values
+    predictions = []
+
+    for label in train_df[id_column].unique():
+        train_sub_df = train_df[train_df[id_column] == label].drop(columns=[id_column], axis=1)
+        train_series = np.array(train_sub_df[dataset.target])
+        train_input = InputData(
+            idx=train_sub_df.index.to_numpy(),
+            features=train_series,
+            target=train_series,
+            task=task,
+            data_type=DataTypesEnum.ts
+        )
+
+        test_sub_df = test_df[test_df[id_column] == label].drop(columns=[id_column], axis=1)
+        test_series = np.array(test_sub_df[dataset.target])
+        test_input = InputData(
+            idx=test_sub_df.index.to_numpy(),
+            features=train_series,
+            target=test_series,
+            task=task,
+            data_type=DataTypesEnum.ts
+        )
+
+        fedot = Fedot(
+            problem=TaskTypesEnum.ts_forecasting.value,
+            task_params=task.task_params,
+            timeout=runtime_min,
+            metric=scoring_metric,
+            seed=config.seed,
+            max_pipeline_fit_time=runtime_min / 10,
+            **training_params
+        )
+
+        with Timer() as training:
+            fedot.fit(train_input)
+        training_duration += training.duration
+
+        with Timer() as predict:
+            try:
+                prediction = fedot.predict(test_input)
+            except Exception as e:
+                log.info('Pipeline crashed. Using no-op forecasting')
+                prediction = np.full(len(test_series), train_series[-1])
+
+        predict_duration += predict.duration
+
+        predictions.append(prediction)
+        models_count += fedot.current_pipeline.length
+
+    save_artifacts(fedot, config)
+    return result(output_file=config.output_predictions_file,
+                  predictions=np.hstack(predictions),
+                  truth=truth_only,
+                  target_is_encoded=False,
+                  models_count=models_count,
+                  training_duration=training_duration,
+                  predict_duration=predict_duration)
```

Comment:
Nit: I think it would be cleaner to remove the line `training_duration, predict_duration = 0, 0` above and just return `training_duration=training.duration, predict_duration=predict.duration` here.

Reply:
since we train the FEDOT model for each series (for each label), `training.duration` value isn't cumulative and will only reflect the time spent on the last iteration

Reply:
My bad, I didn't realize this was happening inside the loop over individual series